app/models/health/qualifying_activity.rb

```diff
@@ -487,16 +455,39 @@ def first_of_type_for_day_for_patient_not_self
       return min_id
     end
 
-    def first_non_care_plan_of_month_for_patient?
-      non_care_plan_for_month_for_patient.minimum(:id) == self.id
+    def first_outreach_of_month_for_patient?
+      outreaches_of_month_for_patient.minimum(:id) == self.id
+    end
+
+    def first_non_outreach_of_month_for_patient?
+      non_outreaches_of_month_for_patient.minimum(:id) == self.id
     end
 
     def number_of_outreach_activities
-      self.class.where(
-        activity: 'outreach',
+      outreaches_by_month = self.class.where(
+        activity: :outreach,
+        patient_id: patient_id,
+        date_of_activity: patient.contributed_enrollment_ranges,
+      ).where(
+        hqa_t[:date_of_activity].lteq(date_of_activity),
+      ).group(
+        Arel.sql("DATE_TRUNC('month', date_of_activity)")
+      ).count
+      outreaches_by_month.reject{|k, v| v.zero?}.keys.count
+    end
+
+    def number_of_non_outreach_activities
+      non_outreaches_by_month = self.class.where(
         patient_id: patient_id,
         date_of_activity: patient.contributed_enrollment_ranges,
+      ).where.not(
+        activity: :outreach,
+      ).where(
+        hqa_t[:date_of_activity].lteq(date_of_activity),
+      ).group(
+        Arel.sql("DATE_TRUNC('month', date_of_activity)")
       ).count
+      non_outreaches_by_month.reject{|k, v| v.zero?}.keys.count
```

Comment:
These will return the count of months with outreach activities yes?

Reply:
Renamed to `number_of_non_outreach_activity_months`.